### ISSUE TITLE
M5-02: Logistic Regression Training

### ISSUE THREAD

**Issue by FabsHof** [FabsHof/diagno_ml#24]
## ✍🏽 Description

Implement model training with Logistic Regression. Contains cross-validation, hyperparameter tuning, and model evaluation.

## 📋 Definition of Done

- [ ] Training script with sklearn LogisticRegression
- [ ] Cross-validation (5-fold stratified)
- [ ] Hyperparameter grid search
- [ ] Evaluation metrics (Accuracy, Precision, Recall, F1, AUC)
- [ ] Confusion matrix visualization
- [ ] Feature importance extraction
- [ ] Model saved as pickle/joblib
- [ ] Integration in Prefect flow

## 👤 User Stories

As a **data scientist**
this feature enables me in training classification models
in order to predict diagnosis probabilities.

## 📝 Hyperparameter Grid

```python
param_grid = {
    'C': [0.001, 0.01, 0.1, 1, 10, 100],
    'penalty': ['l1', 'l2'],
    'solver': ['liblinear', 'saga'],
    'class_weight': [None, 'balanced']
}
```

## ℹ️ Meta Data

- depends_on: M5-01
